### ISSUE TITLE
`NegatedConditionsWithElse` refactoring warning triggered for `!!` operator

### ISSUE THREAD

**Issue by gitneko** [rrrene/credo#1299]
The check `Credo.Check.Refactor.NegatedConditionsWithElse` generates a refactoring warning if you use `!!` in an `if-else` block, despite not actually being a negated condition.

The following code generates the warning:

```ex
defmodule Test do
  def test(a) do
    if !!a do
      1
    else
      2
    end
  end
end
```

Warning:
```
  Refactoring opportunities                                                                                                                                                                                            
┃ 
┃ [F] ↗ Avoid negated conditions in if-else blocks.
┃       lib/test.ex:3 #(Test.test)

Please report incorrect results: https://github.com/rrrene/credo/issues

Analysis took 2.9 seconds (1.0s to load, 1.8s running 75 checks on 233 files)
3177 mods/funs, found 1 refactoring opportunity.

Use `mix credo explain` to explain issues, `mix credo --help` for options.
gg@O72HX892:/mnt/elixir_projekte/bacstack$ mix credo --strict --all --ignore todo --verbose
Checking 233 source files (this might take a while) ...

  Refactoring opportunities                                                                                                                                                                                            
┃ 
┃ [F] ↗ Avoid negated conditions in if-else blocks. [Credo.Check.Refactor.NegatedConditionsWithElse]
┃       lib/test.ex:3 #(Test.test)
┃ 
┃       if !!a do
┃ 

Please report incorrect results: https://github.com/rrrene/credo/issues
```

Since the code is not actually a negated condition, no warning should be generated. The `!!` could be removed in the code, since it's equivalent, but Credo should still not warn about `!!`.
